### PR TITLE
Implement pagination for storyTranslations on frontend

### PR DESCRIPTION
**Pull request by Puepis**
## Notion ticket link

[Paginate getStoryTranslations](https://www.notion.so/uwblueprintexecs/Paginate-getStoryTranslations-d86ce7ed0c904d469226b785af549c98)

## Implementation description
- created a reusable hook to encapsulate logic for handling pagination results
- used the hook to paginate `storyTranslations`
- created UI component to control moving between pages
- removed duplicate apollo client in `index.tsx` 

## Steps to test

1. Log into `planetread+angelamerkel@uwblueprint.org`
2. Go to `/admin`
3. Verify that the story translations table shows 10 items on the first page (and that it shows "1-10 of 13")
4. Click the "next page" button and verify that the remaining items are fetched and displayed properly
5. Apply the "German" language filter and verify that the correct items are displayed
6. Remove the filter and verify that the original results are displayed
7. Play around with adding/removing filters and searching
8. Delete one of the story translations and check that the results are updated properly

## What should reviewers focus on?

- Verify that the previous/next page buttons are enabled/disabled appropriately
- Verify that the results are displayed correctly, no weird behaviour
- Does the size/padding of the pagination UI component look good?
- Does the usePagination hook code make sense?
- Does authentication still work properly (login)

### Some resources for understanding pagination!
- https://www.apollographql.com/blog/graphql/pagination/understanding-pagination-rest-graphql-and-relay/
- https://www.apollographql.com/docs/react/pagination/core-api/

## Checklist

- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] For backend changes, I have run the appropriate linters: `docker exec -it planet-read_py-backend_1 /bin/bash -c "black . && isort --profile black ."` and I have generated new migrations: `flask db migrate -m "<your message>"`
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: frontend/src/components/admin/StoryTranslationsTable.tsx [hunk 1]
```diff
@@ -150,6 +155,9 @@ const StoryTranslationsTable = ({
           }
         />
       )}
+      <TableCaption mt={1} pr={0}>
```
Comment (jennifer-lu) [inline frontend/src/components/admin/StoryTranslationsTable.tsx:158]: Nit: can use marginTop and paddingRight for readability.

File: frontend/src/components/utils/Paginator.tsx [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  last,
+  total,
+  hasPreviousPage,
+  hasNextPage,
+  onPreviousPage,
+  onNextPage,
+}: PaginatorProps) => {
+  return (
+    <Flex justify="flex-end" align="center">
+      <Text mr={2}>
```
Comment (jennifer-lu) [inline frontend/src/components/utils/Paginator.tsx:27]: Nit: can use marginRight for readability.

**Review by jennifer-lu — APPROVED** [review]
Looks good to me! Everything is working as expected (might want to have one other reviewer though because I'm not the most familiar with the code).

File: frontend/src/utils/hooks/usePagination.tsx [hunk 3]
```diff
@@ -0,0 +1,113 @@
+import { DocumentNode, useQuery } from "@apollo/client";
```
Comment (gaoxk) [inline frontend/src/utils/hooks/usePagination.tsx:1]: noiceee custom hook is pretty big brained 😎

File: frontend/src/utils/hooks/usePagination.tsx [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+interface PageInfo {
+  endCursor: string;
+  hasNextPage: boolean;
+}
+
+/*
+ * A hook that adapts cursor-style pagination to support a table-based use-case.
+ * @param query - The paginated GraphQL query
```
Comment (gaoxk) [inline frontend/src/utils/hooks/usePagination.tsx:24]: nit: is it worth mentioning it's relationship with relay?

**Review by gaoxk — APPROVED** [review]
feel free to merge after nits!

File: frontend/src/utils/hooks/usePagination.tsx (hunk 4 above)
Comment (Puepis) [reply frontend/src/utils/hooks/usePagination.tsx:24]: sure, i'll modify the comment!

**Pull request merged**